packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -550,7 +552,7 @@ function commitLifeCycles(
     case ClassComponent: {
       const instance = finishedWork.stateNode;
       if (finishedWork.effectTag & Update) {
-        if (current === null) {
+        if (current === null || finishedWork.effectTag & Placement) {
```

Comment:
Not sure if `Placement` is the right tag to use here or if there's something better.